csi/node_server.go

```diff
@@ -500,16 +520,29 @@ func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 		return nil, status.Error(codes.InvalidArgument, "volume id missing in request")
 	}
 
+	mounter := mount.New("")
+
 	// CO owns the staging_path so we only unmount but not remove the path
-	if err := unmount(targetPath, mount.New("")); err != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("failed to unmount volume %s mount point %v error %v", volumeID, targetPath, err))
+	if err := unmount(targetPath, mounter); err != nil {
+		return nil, status.Error(codes.Internal, errors.Wrapf(err, "failed to unmount volume %s mount point %v", volumeID, targetPath).Error())
```

Comment:
For block volumes, `targetPath` is the parent dir instead then umounting the dir will error out. Hence, should we try the below umounting when `err != nil`?

Reply:
If `targetPath` is not a mount point, the `umount` doesn't work and the error `nil` will be returned.
https://github.com/longhorn/longhorn-manager/blob/master/csi/util.go#L290